cpp/src/tests/connect/con_comp_test.cu

```diff
@@ -0,0 +1,180 @@
+// -*-c++-*-
+
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * NVIDIA CORPORATION and its licensors retain all intellectual property
+ * and proprietary rights in and to this software, related documentation
+ * and any modifications thereto.  Any use, reproduction, disclosure or
+ * distribution of this software and related documentation without an express
+ * license agreement from NVIDIA CORPORATION is strictly prohibited.
+ *
+ */
+
+// connected components tests
+// Author: Andrei Schaffer aschaffer@nvidia.com
+
+//#define DEBUG
+
+#define _DEBUG_C2C
+
+#include "gtest/gtest.h"
+#include "high_res_clock.h"
+#include <cudf.h>
+#include "cuda_profiler_api.h"
+
+#include <cugraph.h>
+#include "test_utils.h"
+#include <algorithm>
+#include <iterator>
+
+// do the perf measurements
+// enabled by command line parameter s'--perf'
+//
+static int PERF = 0;
+
+namespace{ //un-nammed
+  struct Usecase
+  {
+    explicit Usecase(const std::string& a) {
+      // assume relative paths are relative to RAPIDS_DATASET_ROOT_DIR
+      const std::string& rapidsDatasetRootDir = get_rapids_dataset_root_dir();
+      if ((a != "") && (a[0] != '/')) {
+	matrix_file = rapidsDatasetRootDir + "/" + a;
+      } else {
+	matrix_file = a;
+      }
+    }
+
+    Usecase(const Usecase& rhs)
+    {
+      matrix_file = rhs.matrix_file;
+    }
+    
+    Usecase& operator = (const Usecase& rhs)
+    {
+      matrix_file = rhs.matrix_file;
+
+      return *this;
+    }
+    const std::string& get_matrix_file(void) const
+    {
+      return matrix_file;
+    }
+  private:
+    std::string matrix_file;
+  };
+  
+}//end un-nammed namespace
+
+struct Tests_Weakly_CC : ::testing::TestWithParam<Usecase>
+{
+  Tests_Weakly_CC() {  }
+  static void SetupTestCase() {  }
+  static void TearDownTestCase() { 
+    if (PERF) {
+     for (unsigned int i = 0; i < weakly_cc_time.size(); ++i) {
+      std::cout <<  weakly_cc_time[i] << std::endl;
+     }
+    } 
+  }
+  virtual void SetUp() {  }
+  virtual void TearDown() {  }
+
+  static std::vector<double> weakly_cc_time;
+
+  void run_current_test(const Usecase& param) {
+    const ::testing::TestInfo* const test_info =::testing::UnitTest::GetInstance()->current_test_info();
+    std::stringstream ss; 
+    std::string test_id = std::string(test_info->test_case_name()) + std::string(".") + std::string(test_info->name()) + std::string("_") + getFileName(param.get_matrix_file())+ std::string("_") + ss.str().c_str();
+    cudaStream_t stream{nullptr};
+
+#ifndef DEBUG_NO_RMM
+    rmmInitialize(nullptr);
+#endif
+
+    int m, k, nnz; //
+    MM_typecode mc;
+     
+    HighResClock hr_clock;
+    double time_tmp;
+
+    FILE* fpin = fopen(param.get_matrix_file().c_str(),"r");
```

Comment:
Need to check fpin is not nullptr first to avoid segmentation fault, if the file is missing, we need to complain about missing file instead of segfaulting.

Reply:
Better yet, I'll use 'std::ifstream` that relies on RAII, rather than acquire-release C-style.

Reply:
Great, and it will be even better if you can check other test files and make the same changes. Other test files are doing the same thing.

Reply:
I might, but not in this PR. I want to keep conflicts to a minimum, not proliferate them.

Reply:
And... I actually _cannot_ use std::ifstream, because some MM primitives take FILE* pointers, and there's no way to get a FILE* out of an std::fstream. So, I'll add the nullptr check and that's that.